homeassistant/components/keba/__init__.py

```diff
@@ -0,0 +1,229 @@
+"""Support for KEBA charging stations."""
+import asyncio
+import logging
+
+from keba_kecontact.connection import KebaKeContact
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "keba"
+SUPPORTED_COMPONENTS = ["binary_sensor", "sensor", "lock"]
+
+CONF_RFID = "rfid"
+CONF_FS = "failsafe"
+CONF_FS_TIMEOUT = "failsafe_timeout"
+CONF_FS_FALLBACK = "failsafe_fallback"
+CONF_FS_PERSIST = "failsafe_persist"
+CONF_FS_INTERVAL = "refresh_interval"
+
+MAX_POLLING_INTERVAL = 5  # in seconds
+MAX_FAST_POLLING_COUNT = 4
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_HOST): cv.string,
+                vol.Optional(CONF_RFID, default="00845500"): cv.string,
+                vol.Optional(CONF_FS, default=False): cv.boolean,
+                vol.Optional(CONF_FS_TIMEOUT, default=30): cv.positive_int,
+                vol.Optional(CONF_FS_FALLBACK, default=6): cv.positive_int,
+                vol.Optional(CONF_FS_PERSIST, default=0): cv.positive_int,
+                vol.Optional(CONF_FS_INTERVAL, default=5): cv.positive_int,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+_SERVICE_MAP = {
+    "request_data": "request_data",
+    "set_energy": "async_set_energy",
+    "set_current": "async_set_current",
+    "authorize": "async_start",
+    "deauthorize": "async_stop",
+    "enable": "async_enable_ev",
+    "disable": "async_disable_ev",
+    "set_failsafe": "async_set_failsafe",
+}
+
+
+async def async_setup(hass, config):
+    """Check connectivity and version of KEBA charging station."""
+    host = config[DOMAIN][CONF_HOST]
+    rfid = config[DOMAIN][CONF_RFID]
+    refresh_interval = config[DOMAIN][CONF_FS_INTERVAL]
+    keba = KebaHandler(hass, host, rfid, refresh_interval)
+    hass.data[DOMAIN] = keba
+
+    # Wait for KebaHandler setup complete (initial values loaded)
+    if not await keba.setup():
+        _LOGGER.warning("Could not find a charging station at %s", host)
```

Comment:
I think error is appropriate since we're failing the component setup as a result.
```suggestion
        _LOGGER.error("Could not find a charging station at %s", host)
```

Reply:
absolutely correct